pkg/build/controller/factory/factory.go

```diff
@@ -335,6 +335,39 @@ func (factory *BuildConfigControllerFactory) Create() controller.RunnableControl
 	}
 }
 
+type PipelineControllerFactory struct {
```

Comment:
why can't we reuse the existing buildconfig controller and just add the logic for creating jenkins there?